internal/server/device/disk.go

```diff
@@ -2886,14 +2886,16 @@ func (d *disk) generateVMAgentDrive() (string, error) {
 	// Include the most likely agent.
 	if util.PathExists(os.Getenv("INCUS_AGENT_PATH")) {
 		var srcFilename string
-		var dstFilename string
+		dstFilename := "incus-agent"
+		imageOS := strings.ToLower(d.inst.ExpandedConfig()["image.os"])
 
-		if strings.Contains(strings.ToLower(d.inst.ExpandedConfig()["image.os"]), "windows") {
+		if strings.Contains(imageOS, "windows") {
 			srcFilename = fmt.Sprintf("incus-agent.windows.%s", d.state.OS.Uname.Machine)
```

Comment:
I see what you mentioned about architecture.

We indeed shouldn't be using `d.state.OS.Uname.Machine` here but instead use `d.inst.Architecture()` to get the correct architecture name for the instance.